homeassistant/components/osoenergy/sensor.py

```diff
@@ -0,0 +1,179 @@
+"""Support for OSO Energy sensors."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from apyosoenergyapi import OSOEnergy
+from apyosoenergyapi.helper.const import OSOEnergySensorData
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfEnergy, UnitOfPower, UnitOfVolume
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+import homeassistant.util.dt as dt_util
+
+from . import OSOEnergyEntity
+from .const import DOMAIN
+
+
+def _get_local_hour(utc_hour: int):
+    """Get the local hour."""
+    now = dt_util.utcnow()
+    now_local = dt_util.now()
+    utc_time = now.replace(hour=utc_hour, minute=0, second=0, microsecond=0)
+    local_hour = dt_util.as_local(utc_time)
+    return local_hour.replace(
+        year=now_local.year, month=now_local.month, day=now_local.day
+    )
+
+
+def _convert_profile_to_local(values):
+    """Convert UTC profile to local."""
+    profile = [None] * 24
+    for hour in range(24):
+        local_hour = _get_local_hour(hour)
+        profile[local_hour.hour] = values[hour]
+
+    return profile
+
+
+@dataclass(frozen=True, kw_only=True)
+class OSOEnergySensorEntityDescription(SensorEntityDescription):
+    """Class describing OSO Energy heater sensor entities."""
+
+    value: Callable[[OSOEnergy], StateType]
```

Comment:
Let's name this `value_fn` since it's a function
```suggestion
    value_fn: Callable[[OSOEnergy], StateType]
```